core/src/main/java/bisq/core/api/CoreWalletsService.java

```diff
@@ -354,7 +352,7 @@ private LegacyAddress getValidBsqLegacyAddress(String address) {
 
     // Returns a Coin for the double amount, or a RuntimeException if invalid.
     private Coin getValidBsqTransferAmount(double amount) {
-        Coin amountAsCoin = parseToCoin(new BigDecimal(amount).toString(), bsqFormatter);
+        Coin amountAsCoin = parseToCoin(Double.toString(amount), bsqFormatter);
```

Comment:
What is the reason to use `double` for any monetary values? In general I have never seen it done and for anything that's dealing with bitcoins I would use `long` and count satoshis.

Reply:
After much discussion on keybase, changed amount type to String in commit 65df9e1503a9a38518ac29544e8a62cd4ff1045a.

A user needs to be able to enter a representation of a decimal value like "0.25" instead of a satoshi value like 25000000.
The server will convert the string to a Coin for use in a transaction.